src/scripts/gym/Gym.ts

```diff
@@ -27,15 +63,18 @@ class Gym {
         public badgeReward: BadgeEnums,
         public moneyReward: number,
         public defeatMessage: string,
-        public requirements: (OneFromManyRequirement | Requirement)[] = [],
+        requirements: (OneFromManyRequirement | Requirement)[] = [],
         public rewardFunction = () => {},
         {
             quest = true,
             achievement = true,
-        }: gymFlags = {}
+        }: gymFlags = {},
+        buttonText = 'Gym'
     ) {
+        super(requirements);
         this.flags.quest = quest;
         this.flags.achievement = achievement;
+        this.buttonText = buttonText;
```

Comment:
Reduce some duplication:
```suggestion
        public buttonText = 'Gym'
    ) {
        super(requirements);
        this.flags.quest = quest;
        this.flags.achievement = achievement;
```

Reply:
Does not work :(